python/ray/autoscaler/_private/commands.py

```diff
@@ -921,7 +921,8 @@ def exec_cluster(config_file: str,
                  override_cluster_name: Optional[str] = None,
                  no_config_cache: bool = False,
                  port_forward: Optional[Port_forward] = None,
-                 with_output: bool = False) -> str:
+                 with_output: bool = False,
+                 _allow_uninitialized_state: bool = False) -> str:
```

Comment:
Looks like `ray attach` defaults this flag to `True`. Why do we default to `False` for `ray exec`?

Reply:
Actually, why do we need to expose this? can this just always be True?

Reply:
This is set to `True` only from the CLI call to `exec_cluster` in `scripts/scripts.py`. This is to ensure that other cases only work on a running cluster (specifically `ray submit`, `ray dashboard` and `ray monitor`).